deepspeech_pytorch/data/utils.py

```diff
@@ -1,38 +1,65 @@
 from __future__ import print_function
 
-import fnmatch
-import io
+import json
 import os
-import subprocess
+from multiprocessing import Pool
+from pathlib import Path
+from typing import Optional
 
+import sox
 from tqdm import tqdm
 
 
-def create_manifest(data_path, output_name, manifest_path, min_duration=None, max_duration=None):
-    file_paths = [os.path.join(dirpath, f)
-                  for dirpath, dirnames, files in os.walk(data_path)
-                  for f in fnmatch.filter(files, '*.wav')]
-    file_paths = order_and_prune_files(file_paths, min_duration, max_duration)
-    os.makedirs(manifest_path, exist_ok=True)
-    with io.FileIO(os.path.join(manifest_path, output_name), "w") as file:
-        for wav_path in tqdm(file_paths, total=len(file_paths)):
-            transcript_path = wav_path.replace('/wav/', '/txt/').replace('.wav', '.txt')
-            sample = os.path.abspath(wav_path) + ',' + os.path.abspath(transcript_path) + '\n'
-            file.write(sample.encode('utf-8'))
-    print('\n')
+def create_manifest(
+        data_path: str,
+        output_name: str,
+        manifest_path: str,
+        num_workers: int,
+        min_duration: Optional[float] = None,
+        max_duration: Optional[float] = None):
+    data_path = os.path.abspath(data_path)
+    file_paths = list(Path(data_path).rglob('*.wav'))
+    file_paths = order_and_prune_files(
+        file_paths=file_paths,
+        min_duration=min_duration,
+        max_duration=max_duration,
+        num_workers=num_workers
+    )
 
+    output_path = Path(manifest_path) / output_name
+    output_path.parent.mkdir(exist_ok=True, parents=True)
 
-def order_and_prune_files(file_paths, min_duration, max_duration):
+    manifest = {
+        'root_path': data_path,
+        'samples': []
+    }
+    for wav_path in tqdm(file_paths, total=len(file_paths)):
+        wav_path = str(wav_path.relative_to(data_path))
+        transcript_path = wav_path.replace('/wav/', '/txt/').replace('.wav', '.txt')
+        manifest['samples'].append({
+            'wav_path': wav_path,
+            'transcript_path': transcript_path
+        })
```

Comment:
I've found a more robust way to do this was the following.

```suggestion
        wav_path = wav_path.relative_to(data_path)
        transcript_path = wav_path.parent.with_name("txt") / wav_path.with_suffix(".txt").name
        manifest['samples'].append({
            'wav_path': wav_path.as_posix(),
            'transcript_path': transcript_path.as_posix()
        })
```

Otherwise, the string replacement might not work with files with certain names.

Reply:
Epic! Thanks so much